tools/individual-packages-gen/gen-phase1.ts

```diff
@@ -0,0 +1,209 @@
+            : 'cdk-alpha.aws-' + jsiiTargets.java.maven.artifactId;
+          jsiiTargets.python.distName = jsiiTargets.python.distName.replace(
+            /^aws-cdk\./, 'aws-cdk.alpha.');
+          jsiiTargets.python.module = jsiiTargets.python.module.replace(
+            /^aws_cdk\./, 'aws_cdk.alpha.');
+
+          // disable awslint (some rules are hard-coded to @aws-cdk/core)
+          srcPackageJson.awslint = {
+            exclude: ['*:*'],
+          };
+
+          // add a pkglint exemption for the 'package name = dir name' rule
+          const pkglint = srcPackageJson.pkglint || {};
+          pkglint.exclude = [
+            ...(pkglint.exclude || []),
+            'naming/package-matches-directory',
+            // the experimental packages need the "real" assert dependency
+            'assert/assert-dependency',
+          ];
+          srcPackageJson.pkglint = pkglint;
+
+          // regular dependencies
+          const alphaDependencies: { [dep: string]: string } = {};
+          for (const dependency of Object.keys(srcPackageJson.dependencies || {})) {
+            // all 'regular' dependencies on alpha modules will be converted to
+            // a pair of devDependency on '0.0.0' and peerDependency on '^0.0.0',
+            // and the package will have no regular dependencies anymore
+            if (alphaPackages[dependency]) {
+              alphaDependencies[alphaPackages[dependency]] = pkg.version;
+            }
+          }
+          srcPackageJson.dependencies = undefined;
+
+          // devDependencies
+          const alphaDevDependencies: { [dep: string]: string } = {
+            // we need jsii-rosetta in the dependencies,
+            // otherwise the binary cannot be found when running the 'extract' script
+            'jsii-rosetta': jsiiRosettaVersion,
+          };
+          const devDependencies = srcPackageJson.devDependencies || {};
+          for (const devDependency of Object.keys(devDependencies)) {
+            if (devDependency.startsWith('@aws-cdk/')) {
+              delete devDependencies[devDependency];
+            }
+            if (alphaPackages[devDependency]) {
+              alphaDevDependencies[alphaPackages[devDependency]] = pkg.version;
+            }
+          }
+          devDependencies['@aws-cdk/assert'] = pkg.version;
+          devDependencies['aws-cdk-lib'] = pkg.version;
+          devDependencies.constructs = '^10.0.0';
+          // we save the devDependencies in a temporary key in package.json
+          srcPackageJson.tmp_devDependencies = {
+            ...devDependencies,
+            ...alphaDevDependencies,
+            ...alphaDependencies,
+          };
+          srcPackageJson.devDependencies = {
+            ...alphaDevDependencies,
+            ...alphaDependencies,
+          };
+
+          // peer dependencies
+          srcPackageJson.tmp_peerDependencies = {
+            'aws-cdk-lib': `^${pkg.version}`,
+            'constructs': '^10.0.0',
```

Comment:
We shouldn't be manipulating dependencies or their versions anywhere here.

We should just change the `package.json` file for these packages in the v2-main branch and add an pkglint rule for it (like this one - https://github.com/aws/aws-cdk/blob/master/tools/pkglint/lib/rules.ts#L629).

Forward merge is already configured to run pkglint and commit the changes, so for any new packages this will be fixed up.

*Why?* - Dependencies should all be managed in one place - `package.json`. Tweaking or changing them somewhere else will bring in unintended side-effects, and make things hard to figure out.

Reply:
This is not possible (these packages do not build with `aws-cdk-lib` as their dependencies, they use `@aws-cdk/*` packages).

Reply:
Remember that `pkglint` runs on these generated packages too, so any violations will be caught at build time.